src/games/oware.rs

```diff
@@ -0,0 +1,247 @@
+//! Oware is a strategy game from the family of mancala
+//!
+//! It has no  combos, i.e. one turn each round each player.
+//! Official version contains 12 pits and 4 initial seeds in each pit. This implementation is const generic parameterized with PITS and INIT_SEEDS.
+//!
+//! # Rules
+//! 1. Each player has control over half the pits that are in front of them
+//! 2. Player has to choose a non-empty pit. This removes the seeds from that pit and
+//!    distributes them, dropping one in each pit, except the selected pit,
+//!    in counter-clock wise.
+//! 3. If the last seed is dropped in one of the opponents pit, and the final seed count in that
+//!    pit is 2 or 3, then the current player captures it. It can happen several times if
+//!    continous multiple pits to the last dropped pit also have `2` or `3` seeds in them.
+//!    If all the seeds could be captured this turn, then no capture happens
+//!
+//! Objective is to capture as many seeds as possible and so the game will conclude early if a
+//! player captures more than half the seeds
+use std::fmt::{Debug, Display, Formatter};
+
+use internal_iterator::{Internal, IteratorExt};
+
+use crate::board::{Alternating, Board, BoardMoves, BruteforceMoveIterator, Outcome, Player};
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct OwareBoard<const PITS: usize, const INIT_SEEDS: u8> {
+    pits: [u8; PITS],
+    scores: [u8; 2],
+    next_player: Player,
+    outcome: Option<Outcome>,
+}
+
+impl<const P: usize, const INIT_SEEDS: u8> Default for OwareBoard<P, INIT_SEEDS> {
+    fn default() -> Self {
+        OwareBoard {
+            pits: [INIT_SEEDS; P],
+            scores: Default::default(),
+            next_player: Player::A,
+            outcome: None,
+        }
+    }
+}
+
+impl<const PITS: usize, const I: u8> OwareBoard<PITS, I> {
+    fn is_in_opp_pits(&self, mv: usize) -> bool {
+        assert!(mv < PITS);
+        mv * 2 / PITS == usize::from(self.next_player == Player::A)
+    }
```

Comment:
The rest of the repo adds newlines between functions, I'm surprised cargo fmt doesn't already enforce this.

Reply:
cargo fmt only enforces upper bound of 1 line between blocks. I will add newlines between.